server.go

```diff
@@ -205,16 +278,116 @@ func callbackHandler(w http.ResponseWriter, r *http.Request, conf config) {
 	http.Redirect(w, r, state, http.StatusTemporaryRedirect)
 }
 
+func refreshCheckHandler(w http.ResponseWriter, r *http.Request, conf *config) {
+	redirectURL := conf.oktaLoginBaseURL + "&prompt=none&state=/sso/refresh/done"
+
+	tokenCookie, err := r.Cookie(conf.cookieName)
+	switch {
+	case err == http.ErrNoCookie:
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Printf("refreshHandler: No Cookie")
+		return
+	case err != nil:
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Printf("refreshHandler: Error parsing cookie,  %v", err)
+		return
+	}
+
+	jwt, err := conf.verifier.VerifyAccessToken(tokenCookie.Value)
+
+	if err != nil {
+		w.WriteHeader(http.StatusUnauthorized)
+		log.Printf("refreshHandler: JWT Validation Error,  %v", err)
+		return
+	}
+
+	exp, ok := jwt.Claims["exp"]
+	if !ok {
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Printf("refreshHandler: Claim 'exp' not included in access token,  %v", tokenCookie.Value)
+		return
+	}
+
+	expFloat, ok := exp.(float64)
+	if !ok {
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Printf("refreshHandler: Unable to convert 'exp' to float64")
+		return
+	}
+
+	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+	if expFloat-float64(time.Now().UTC().Unix()) < (5 * time.Minute).Seconds() {
+		io.WriteString(w, redirectURL)
+	} else {
+		io.WriteString(w, "ok")
+	}
+}
+
+func refreshDoneHandler(w http.ResponseWriter, r *http.Request, conf *config) {
+	w.Header().Set("Content-Type", "text/html; charset=utf-8")
+	io.WriteString(w, `
+	<!DOCTYPE html>
+	<html>
+		<head>
+			<meta charset="UTF-8">
+			<title>SSO Refresh</title>
+			<script>
+				window.parent.postMessage("ssoRefreshDone","`+conf.appOrigin+`");
+			</script>
+		</head>
+		<body>
+			SSO Refresh
+		</body>
+	</html> 
+	`)
+}
+
+func errorHandler(w http.ResponseWriter, r *http.Request, conf *config) {
+	params := r.URL.Query()
+	ssoErr := params.Get("error")
+
+	w.WriteHeader(http.StatusUnauthorized)
+
+	io.WriteString(w, `
+	<!DOCTYPE html>
```

Comment:
Can we put this in a "const"?

Reply:
I don't think so, the `conf.appOrigin` variable is part of the string so it can't be const